soc/xtensa/intel_adsp/common/boot.c

```diff
@@ -69,6 +69,9 @@ __asm__(".section .imr.z_boot_asm_entry, \"x\" \n\t"
 	"  movi  a0, 0x4002f        \n\t"
 	"  wsr   a0, PS             \n\t"
 	"  movi  a0, 0              \n\t"
+#ifdef XCHAL_HAVE_THREADPTR
+	"  wur   a0, THREADPTR      \n\t"
+#endif
```

Comment:
Can you move this to an `__asm__` block in the C code instead?  This block here is intended to be the absolute minimum amount of bootstrapping (really just the PS and register window state) required to enter a C function.  Long files of promiscuous assembly become technical debt really fast.

Reply:
It can ... but I am not sure that is cleaner tbh. That will be required in very early stages of the soc initialization (while still running from IMR) and moving it for the C code will make it more spaghetti IMHO.

Reply:
Trust me, it's easier.  There's even a spot already there for it shared between cores: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/xtensa/intel_adsp/common/include/cpu_init.h#L21

Reply:
Yep, I am convinced :) That is a better place indeed.